File: atcoder/benchmark/golang/benchmark.go
package main

import (
	"fmt"
	"math"
	"time"
)
import "github.com/dustin/go-humanize"

func main() {
	for i := 4; i <= 9; i++ {
		benchmark_op(int(math.Pow10(i)))
	}
}

/*
 * 複数行コメントは使われていない?
 */
func benchmark_op(N int) {
	humanized := humanize.Comma(int64(N))
	fmt.Printf("benchmark, N:%s\n", humanized)
	var a int
	a = 0

	d := time.Now()
	for i := 0; i <= N; i++ {
		// nothing
	}
	fmt.Println("for", time.Now().Sub(d))

	d = time.Now()
	for i := 1; i <= N; i++ {
		a += i + 10000000
	}
	fmt.Println("add", time.Now().Sub(d))

	d = time.Now()
	for i := 1; i <= N; i++ {
		a += i - 10000000
	}
	fmt.Println("Sub", time.Now().Sub(d))

	d = time.Now()
	for i := 1; i <= N; i++ {
		a += i * i
	}
	fmt.Println("mul", time.Now().Sub(d))

	d = time.Now()
	for i := 1; i <= N; i++ {
		a += 10000000 / i
	}
	fmt.Println("div", time.Now().Sub(d))

	fmt.Println("")
}

File: atcoder/benchmark/python3/benchmark.py
import sys
import time


def main():
    # 10 * 1000 * 1000
    pow = 7
    attr_pypy = "pypy_version_info"
    if hasattr(sys, attr_pypy):
        print(sys.version)
        pow = 8

    for _ in range(4, pow + 1):
        benchmark_op(10 ** _)


def benchmark_op(N):
    print("benchmark op(ms), N:{:,d}".format(N))
    a = 0

    t = time.time()
    for i in range(1, N + 1):
        pass
    print("for", "{:.10f}".format((time.time() - t) * 1000), sep='\t')

    t = time.time()
    for i in range(1, N + 1):
        a += i + 10000000
    print("add", "{:.10f}".format((time.time() - t) * 1000), sep='\t')

    t = time.time()
    for i in range(1, N + 1):
        a += i - 10000000
    print("sub", "{:.10f}".format((time.time() - t) * 1000), sep='\t')

    t = time.time()
    for i in range(1, N + 1):
        a += i * i
    print("mul", "{:.10f}".format((time.time() - t) * 1000), sep='\t')

    t = time.time()
    for i in range(1, N + 1):
        a += 10000000 / i
    print("div", "{:.10f}".format((time.time() - t) * 1000), sep='\t')

    print("")

if __name__ == '__main__':
    main()
